### ISSUE TITLE
Optimise the screenshot function

### ISSUE THREAD

**Issue by Technerder** [sp614x/optifine#2067]
When taking a screenshot in versions from 1.7 and up, the screenshot function takes too long and temporarily freezes the game. This is because the screenshot method saves the screenshot to a file at the same time on the same thread as the one rendering the game display.

**Comment by Redspyro99** [comment]
Recently I've been taking screenshots in Minecraft and I've noticed that occasionally my game would freeze when taking a screenshot just as mentioned, but occasionally I'd notice it wouldn't freeze up for a good long while. I just ran some testing just now and I think I found something that could be of interest. The game doesn't seem to lock up when Fast Rendering is Disabled, while it does freeze up when it's Enabled.  Now perhaps I'm just an outlier, but maybe that could help with solving this problem.

**Comment by sp614x** [comment]
The screenshot is taken in a different way when Fast Render is enabled. This is why custom screenshot size doesn't work with Fast Render. Both ways should be identically fast, but different graphics drivers may behave differently.

**Comment by Redspyro99** [comment]
Yeah, it does appear to be driver dependent. Although I wonder if it also depends on the graphics hardware as well. I could only run two tests, one with a GTX 1050 with 417.22 drivers and an AMD Radeon HD 8330 integrated GPU running 16.12.2 drivers. The 1050 experienced the freeze when Fast Render was enabled, while the 8330, didn't experience the freeze. I have no clue if any of this would help at all, but I suppose it's worth a shot anyways.
